projects/igniteui-angular/migrations/common/ServerHost.ts

```diff
@@ -38,15 +38,24 @@ export class ServerHost implements ts.server.ServerHost {
         return ts.sys.getFileSize(path);
     }
 
-    public watchFile(path: string, callback: ts.FileWatcherCallback, pollingInterval?: number):
+    //#region Watchers
+    // Atm we do not need to have file or dir watchers that access the actual FS
+    // as we are only working with NG's virtual FS
+    // Prior to https://github.com/microsoft/TypeScript/pull/49990 we were more or less required
+    // to add a watcher since it threw an error otherwise
+
+    public watchFile(_path: string, _callback: ts.FileWatcherCallback, _pollingInterval?: number):
         ts.FileWatcher {
-        return ts.sys.watchFile(path, callback, pollingInterval);
+        // return ts.sys.watchFile(path, callback, pollingInterval);
+        return undefined;
```

Comment:
PS: This relies a little bit on the internal TypeScript handling, which is not guaranteed in the future. Would've preferred if we still return something that still satisfies the public interface (`watchFile`'s signature doesn't expect undefined result), e.g. a simple mock with noop close method - just to be safe. Regardless, if we monitor this regularly (or if automation covers it) this should be okay for now.

Reply:
And just for ref, happened as expected and the undefined watcher is now checked and throws so this resolved it https://github.com/IgniteUI/igniteui-angular/pull/13478/commits/f6fd00e4c3ca84157ae54026ebeb71fe8ef7281a